docs/modules/ROOT/pages/lua/examples/paint.adoc

```diff
@@ -1,77 +1,187 @@
+( Red , Green , Blue , White , Transparent )
 
-This small program uses a GPU and a Screen to make a small paint program ingame.
+You can draw on the screen with the selected  
+color by holding down your primary mouse  
+button and moving your mouse around.
 
-In the lower left corner you can choose between five different colors (clear, red, green, blue and white)
-by simply clicking with your mouse onto one.
+== Large Screen Setup
 
-After that you can begin to draw on the screen with that selected color, by click and holding the mouse button
-and then moving the mouse around. To stop drawing simply release the mouse button.
+image::lua/examples/Example_Paint_Large-Screen_Hardware.png[ width = 300 , height = 300 , role=right ]
 
-== Setup
+1. Place a computer case with:
+    - Tier 1 CPU 
+    - 1 stick of RAM 
+    - Tier 1 GPU
 
-You need small computer with a Lua CPU, a little bit RAM, a Screen Driver and GPU T1.
-Then just plug in a Lua EEPROM and fill it with the code below.
-Then just run it and switch to the computer screen and have some fun ^^
+2. Place & connect a large screen
 
-== Code
+3. Plug in an EEPROM
+
+4. Paste the setup code + program code
+
+5. Start the computer
 
 [source,Lua]
 ----
--- get first T1 GPU avialable from PCI-Interface
-local gpu = computer.getPCIDevices(findClass("GPUT1"))[1]
-if not gpu then
- error("No GPU T1 found!")
-end
+local gpu = computer.getPCIDevices(classes.GPU_T1_C)[1]
 
--- get first Screen-Driver available from PCI-Interface
-local screen = computer.getPCIDevices(findClass("FINComputerScreen"))[1]
--- if no scren found, try to find large screen from component network
-if not screen then
- local comp = component.findComponent(findClass("Screen"))[1]
- if not comp then
-  error("No Screen found!")
- end
- screen = component.proxy(comp)
-end
+assert(gpu,'No <GPU_T1_C> PCI device found')
+
+
+local comp_screen = component.findComponent(classes.Screen)[1]
+
+assert(comp_screen,'No <Screen> component found')
+
+
+local screen = component.proxy(comp_screen)
+
+assert(screen,'Failed to proxy <Screen> component')
+----
+
+
+== Internal Screen Setup
+
+image::lua/examples/Example_Paint_Internal-Screen_Hardware.png[ width = 300 , height = 300 , role = right ]
+
+1. Place a computer case with:
+    - Tier 1 CPU 
+    - 1 stick of RAM 
+    - Screen Driver 
+    - Tier 1 GPU
+
+2. Plug in an EEPROM
+
+3. Paste the setup code + program code
+
+4. Start the computer
+
+5. Switch to the screen tab
+
+[source,Lua]
+----
+local gpu = computer.getPCIDevices(classes.GPU_T1_C)[1]
+
+assert(gpu,'No <GPU_T1_C> PCI device found')
+
+
+local screen = computer.getPCIDevices(classes.FINComputerScreen)[1]
+
+assert(screen,'No <FINComputerScreen> PCI device found')
+----
+
+== Program Code
+
+
+[source,Lua]
+----
+-- [ GPU Setup ]
 
--- setup gpu
 event.listen(gpu)
 gpu:bindScreen(screen)
-w, h = gpu:getSize()
 
--- clear background
+local width , height = gpu:getSize()
+
+print('Screen Size',width,'x',height)
```

Comment:
The final example is still missing the combined approach as discussed.
1 Code Sample you can C&P and run... that works with Large Screen and Internal Screen.

Reply:
Oh fuck sorry, I'll get to it~